jog_arm/src/jog_arm/jog_calcs.cpp

```diff
@@ -554,28 +557,25 @@ namespace jog_arm {
 
 // Parse the incoming joint msg for the joints of our MoveGroup
   bool JogCalcs::updateJoints() {
-    // Check if every joint was zero. Sometimes an issue.
-    bool all_zeros = true;
-
     // Check that the msg contains enough joints
     if (incoming_jts_.name.size() < jt_state_.name.size())
       return false;
 
     // Store joints in a member variable
     for (std::size_t m = 0; m < incoming_jts_.name.size(); ++m) {
-      for (std::size_t c = 0; c < jt_state_.name.size(); ++c) {
-        if (incoming_jts_.name[m] == jt_state_.name[c]) {
-          jt_state_.position[c] = incoming_jts_.position[m];
-          // Make sure there was at least one nonzero value
-          if (incoming_jts_.position[m] != 0.)
-            all_zeros = false;
-          goto NEXT_JOINT;
-        }
+      std::size_t c;
+      try {
+        c = jt_state_name_map_.at(incoming_jts_.name[m]);
       }
-      NEXT_JOINT:;
+      catch(const std::out_of_range& e) {
+        ROS_ERROR_STREAM_NAMED(NODE_NAME, "Command joint name unknown.");
+        return false;
+      };
+
+      jt_state_.position[c] = incoming_jts_.position[m];
     }
 
-    return !all_zeros;
+    return true;
```

Comment:
Unfortunately this breaks the check for all-zero incoming commands -- which means the jogger publishes 0 velocities all the time and I can't execute trajectories anymore.